ml_model/model/classification/main.py

```diff
@@ -0,0 +1,544 @@
+    f1_cw_overall = 2 * (precision_cw_overall * recall_cw_overall) / (precision_cw_overall + recall_cw_overall)
+    accuracy_overall_cw = (tp_cw_overall + tn_cw_overall) / (tp_cw_overall + tn_cw_overall + fp_cw_overall + fn_cw_overall)
+
+    return tp_c_overall, tn_c_overall, fp_c_overall, fn_c_overall, precision_c_overall, recall_c_overall, f1_c_overall, tp_cw_overall, tn_cw_overall, fp_cw_overall, fn_cw_overall, precision_cw_overall, recall_cw_overall, f1_cw_overall,accuracy_overall_c ,accuracy_overall_cw, mean_auc_cw_c, mean_auc_c_c, mean_auc_cw_cw, mean_auc_c_cw
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Run classification models")
+    # debug parameters
+    parser.add_argument(
+        "--debug", type=int, choices=[0, 1], default=0,
+        help="debug mode, output extra info & break all loops." "0: disable, 1 enable")
+    parser.add_argument(
+        "--folds", type=int, default=10,
+        help="number of folds for cross validation (optional)")
+    parser.add_argument(
+        "--output_file", type=str, default=output_file,
+        help="output file name (optional)")
+
+    args = parser.parse_args()
+    folds = args.folds
+    output_file = args.output_file
+    print("Processing with {} folds.".format(args.folds) if args.folds else "Processing without folds.")
+
+    if args.debug:
+        LOGGER.setLevel(logging.DEBUG)
+        LOGGER.debug("Debug mode enabled")
+
+    df = pd.read_csv(ROOT_PATH + "data/understandability_with_warnings.csv")
+
+    ## write header
+    with open(ROOT_PATH + "Results/" + output_file, "w") as csv_file:
+        writer = csv.DictWriter(csv_file, fieldnames=dict_data.keys())
+        writer.writeheader()
+
+    ## read json file
+    with open(ROOT_PATH + "classification/experiments.jsonl") as jsonl_file:
+        experiments = [json.loads(jline) for jline in jsonl_file.read().splitlines()]
+        
+        model_names = ["SVC", "knn_classifier", "logistic_regression", "randomForest_classifier", "mlp_classifier"]
+        
+        for model_name in model_names:
+            for experiment in experiments:
+                ## drop rows with missing values in the feature
+                full_dataset = df.dropna(subset=experiment["target"])
+                target_y = full_dataset[experiment["target"]]
+                
+                ## StratifiedKFold
+                kf = StratifiedKFold(n_splits=folds, shuffle=True, random_state=42)
+                # kf = LeaveOneOut()
+                
+                ## for code + warning features
+                feature_X_cw = full_dataset[experiment["features"]]
+                
+                ## for code features
+                # drop the frist 1 column
+                feature_X_c = full_dataset[experiment["features"]].iloc[:, 1:]
+                
+
+                ite = 1
+
+                ## keep track of the best model for each fold
+                best_hyperparams_c = set()
```

Comment:
We just need one set of hyperparameters because there is no need/purpose for both